mbuild/formats/gomc_conf_writer.py

```diff
@@ -0,0 +1,5026 @@
+                    or os.path.splitext(self.Structures_box_0)[1] != '.psf' \
+                    or os.path.splitext(self.Coordinates_box_1)[1] != '.pdb' \
+                    or os.path.splitext(self.Structures_box_1)[1] != '.psf':
+                warn('ERROR: The proper force field, PDB, and psf files were not provided, '
+                     'or at least their extentions are not correct '
+                     '(i.e., not .inp, .pdb, or .psf). Or box 1 PSF and PDB files were not provided '
+                     'for the GEMC_NVT, GEMC_NPT or GCMC simulations, which is not allowed')
+                self.input_error = True
+                print_error_message = 'ERROR: The proper force field, PDB, and psf files were not provided, '\
+                                      'or at least their extentions are not correct '\
+                                      '(i.e., not .inp, .pdb, or .psf). Or box 1 PSF and PDB files were not provided '\
+                                      'for the GEMC_NVT, GEMC_NPT or GCMC simulations, which is not allowed'
+                raise ValueError(print_error_message)
+        else:
+            print("INFO: All the required force field, pdb, and psf files for box 0 and 1 (.inp, .pdb, and .psf) all "
+                  "passed the intial error checking. Note: the file names and their existance is not confirmed.")
+
+        if input_variables_dict is None:
+            self.input_variables_dict = {}
+        elif isinstance(input_variables_dict, dict) is True:
+            self.input_variables_dict = input_variables_dict
+        else:
+            self.input_error = True
+            print_error_message = "ERROR: The input_variables_dict variable is not None or a dictionary. "
+            raise ValueError(print_error_message)
+
+        # Create all lower case spelled keywords, and return case specific keywords
+        # Also, creates a dict to convert the lower case keys to case sensitive keys
+        all_input_var_case_spec_list = _get_all_possible_input_variables(description=False)
+        all_input_var_case_unspec_list = []
+        all_input_var_case_unspec_to_spec_dict = {}
+        for var_i in all_input_var_case_spec_list:
+            all_input_var_case_unspec_list.append(var_i.lower())
+            all_input_var_case_unspec_to_spec_dict.update({var_i.lower(): var_i})
+
+        # create/fix user case insensitive input variables (input_variables_dict) keys to case sensitive keys
+        input_var_dict_orig_keys_list = dict_keys_to_list(self.input_variables_dict)
+        for z_j in range(0, len(input_var_dict_orig_keys_list)):
+            key_lower = input_var_dict_orig_keys_list[z_j].lower()
+            if key_lower in all_input_var_case_unspec_list:
+                input_variables_dict[all_input_var_case_unspec_to_spec_dict[key_lower]] = \
+                    input_variables_dict.pop(input_var_dict_orig_keys_list[z_j])
```

Comment:
Why not fix this while assigning the properties to this class [here](https://github.com/mosdef-hub/mbuild/pull/831/files#diff-0b2e5a877443b88f11f1adea5e53a392b7b43da2ea7cd59ed0f3f7b88e32c73eR1800)

Reply:
The gom_conf_writer.py code section that changes any cased user input key to the proper cased key is moved higher up in the code, at the bottom of the variable assignments.